### ISSUE TITLE
Overriding endpoints with RTK Query and Vite HMR

### ISSUE THREAD

**Issue by kendallroth** [reduxjs/redux-toolkit#3576]
I am using Redux Toolkit Query in a Vite project, and getting the typical warnings about overriding endpoints when the store HMR reloads.

```
called 'injectEndpoints' to override already-existing endpointName {NAME} without specifying 'overrideExisting: true'
```

As seen in this [GitHub issue](https://github.com/reduxjs/redux-toolkit/issues/3157), a potential fix was suggested to set `overrideExisting` during a Webpack HMR reload. This would sidestep the issue by allowing endpoints to be overridden "silently" when expected (ie. during HMR reload). However, this fix does not work in Vite, as Vite HMR operates differently. I am hoping to get a fix for Vite's HMR workflow, but have been unable to find anything anywhere.

```
// Add to `injectEndpoints` call
overrideExisting: module.hot?.status() === "apply"
```

How can I only enable overriding existing RTK Query endpoints when the refresh/override was triggered by **Vite** HMR?

_P.S. Not overriding existing endpoints certainly explains why I often need to hard refresh to actually notice changes made to RTK Query endpoints..._

**Comment by markerikson** [comment]
I don't think we have a particular answer here - none of us is working on a Vite project atm, and we don't know anything about how Vite HMR usage differs from Webpack.

**Comment by phryneas** [comment]
You could try something like `process.env.NODE_ENV == "development"` though.

**Comment by kendallroth** [comment]
> I don't think we have a particular answer here - none of us is working on a Vite project atm, and we don't know anything about how Vite HMR usage differs from Webpack.

That's definitely fair, and I may instead ask this within Vite, as it is arguably more geared towards the HMR process (as compared to RTK). Thanks for the quick response!

> You could try something like `process.env.NODE_ENV == "development"` though.

Yeah, that's what I'm currently using (or rather, `overrideExisting: import.meta.env.DEV`), but it does _technically_ mean that should an endpoint be defined twice (through non-HMR means), it wouldn't get caught in development 🤷. However, it _certainly_ is a semi-acceptable workaround to get away from the console warnings 👍.
